server/client.go

```diff
@@ -1541,6 +1541,10 @@ func (c *client) processConnect(arg []byte) error {
 	if ws := c.ws; ws != nil && c.opts.JWT == "" {
 		c.opts.JWT = ws.cookieJwt
 	}
+	// when not in operator mode, discard the jwt
+	if srv != nil && srv.trustedKeys == nil {
```

Comment:
Right, I think it is immutable for now, but otherwise we could capture if we are in operator mode at top of function (What we can't have is c-lock -> s-lock). Of course, since this is a client receiver, we need c.srv, which then technically would need to be protected, but again, I think that one is immutable.
By the way, there is a call at the top that assumes that c.srv is not nil: `supportsHeaders := c.srv.supportsHeaders()`.

Reply:
mentation for supportsHeaders is such that it returns false if the receiver is nil.

Reply:
@kozlovic is it ok to access c.srv without a lock as c.srv.supportsHeaders() does?

Reply:
Outside of test, I believe it is set when the client is created and again I don't think it is set to nil. So we could have the same function than for supportsHeaders() that is friendly to c.srv being nil, but then acquire the server lock to return if trusted keys are empty or not.
But again, since right now trustedKeys is immutable, let leave this as-is.